ykutil/src/addr.rs

```diff
@@ -1,21 +1,35 @@
 //! Address utilities.
 
 use libc::{c_void, dladdr, Dl_info};
-use once_cell::sync::Lazy;
 use phdrs::objects;
 use std::{
     convert::TryFrom,
     env,
-    ffi::{CStr, CString},
+    ffi::CStr,
+    fs,
+    path::{Path, PathBuf},
     ptr::{null, null_mut},
 };
 
-static CUR_EXE_C: Lazy<CString> =
-    Lazy::new(|| CString::new(env::current_exe().unwrap().to_str().unwrap()).unwrap());
+// Make the path `p` a canonical absolute path.
+//
+// PERF: Cache the results to avoid file-system usage.
+//
+// PERF: A lot of the canonicalisation work we do is for the object names in the program headers.
```

Comment:
I think this is the right `PERF` and we should delete the "cache" `PERF` above.

Reply:
The second `PERF` doesn't cover all canonicalizations, hence both comments.

(The other case is [here](https://github.com/ykjit/yk/pull/367/files/03b373c8bdb38bcbcf1af170a213bfadb42028b2..d7dcbe3f667db477b1291f74d61be7745ce2aa6f#diff-aa389a21efb030fbbaf6bba425c697d7076980ec84b58822881b2acc78d3778cR43))

Reply:
I'm actually pretty nervous about this function because caching can give different results than non-caching (if someone moves files around). I worry that this function will bite us on the bottom if we ever make it cache. Can we fix the second case another way and/or maybe it's not actually a bottleneck?

Reply:
I don't want to get too bogged down in optimisation now, but I think {re,}moving binaries and libraries will likely be fatal for our system anyway. For example the PT decoder relies on on-disk objects to construct a trace, and we need to query objects for their symbols etc.

For now shall we can weaken our `PERF` comment to something like `PERF: Repeated canonicalisation is likely to be slow. Is caching beneficial?`. Then our future selves can have fun profiling.

Reply:
I'd rather not mention caching at all and I'd rather not implement caching in the future unless this turns out to be absolutely killer to performance.

Reply:
Fair enough then. Removed.